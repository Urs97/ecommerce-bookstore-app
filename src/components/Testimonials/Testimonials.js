import React from "react";
import {
  CarouselProvider,
  Slider,
  Slide,
  ButtonBack,
  ButtonNext,
} from "pure-react-carousel";
import "pure-react-carousel/dist/react-carousel.es.css";
import "./Testimonials.scss";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAngleLeft } from "@fortawesome/free-solid-svg-icons";
import { faAngleRight } from "@fortawesome/free-solid-svg-icons";

import portrait_1_webp from "../../assets/images/testimonial-portrait-1.webp";
import portrait_2_webp from "../../assets/images/testimonial-portrait-2.webp";
import portrait_3_webp from "../../assets/images/testimonial-portrait-3.webp";
import portrait_1_jpg from "../../assets/images/testimonial-portrait-1.jpg";
import portrait_2_jpg from "../../assets/images/testimonial-portrait-2.jpg";
import portrait_3_jpg from "../../assets/images/testimonial-portrait-3.jpg";

class Testimonials extends React.Component {
  render() {
    const text_1 =
      "I like Bookz more and more each day because it makes my life a lot easier. I'd really be lost without Bookz.";
    const text_2 =
      "I found what I needed and books arrived in a timely manner. The prices are super low, keep up the good work!";
    const text_3 =
      "It really saves me time and effort. Accesible book reading is exactly what I feel has been lacking so far and I have nothing but praise for the way Bookz goes about it!";
    const name_1 = "Daniel Smith";
    const name_2 = "Holly Dangerfield";
    const name_3 = "Steve Abrams";

    return (
      <CarouselProvider
        className="testimonial-provider"
        naturalSlideWidth={16}
        naturalSlideHeight={9}
        totalSlides={3}
        isPlaying={true}
        infinite={true}
      >
        <Slider className="testimonial">
          <Slide className="testimonial-slide">
            <div className="wrapper">
              <picture>
                <source type="image/webp" srcSet={`${portrait_1_webp} 768w`} />
                <source type="image/jpg" srcSet={`${portrait_1_jpg} 768w`} />
                <img
                  src={portrait_1_jpg}
                  alt="Facial portrait by Štefan Štefančík on Unsplash"
                />
              </picture>
              <p>{text_1}</p>
              <span>{name_1}</span>
            </div>
          </Slide>
          <Slide className="testimonial-slide">
            <div className="wrapper">
              <picture>
                <source type="image/webp" srcSet={`${portrait_2_webp} 768w`} />
                <source type="image/jpg" srcSet={`${portrait_2_jpg} 768w`} />
                <img
                  src={portrait_2_jpg}
                  alt="Facial portrait by Robert Godwin on Unsplash"
                />
              </picture>
              <p>{text_2}</p>
              <span>{name_2}</span>
            </div>
          </Slide>
          <Slide className="testimonial-slide">
            <div className="wrapper">
              <picture>
                <source type="image/webp" srcSet={`${portrait_3_webp} 768w`} />
                <source type="image/jpg" srcSet={`${portrait_3_jpg} 768w`} />
                <img
                  src={portrait_3_jpg}
                  alt="Facial portrait by Mahesh Krishnamurthy on Unsplash"
                />
              </picture>
              <p>{text_3}</p>
              <span>{name_3}</span>
            </div>
          </Slide>
        </Slider>
        <ButtonBack className="testimonials-btn">
          <span role="img" aria-label="left arrow icon">
            <FontAwesomeIcon icon={faAngleLeft} />
          </span>
        </ButtonBack>
        <ButtonNext className="testimonials-btn tstm-btn-next">
          <span role="img" aria-label="right arrow icon">
            <FontAwesomeIcon icon={faAngleRight} />
          </span>
        </ButtonNext>
      </CarouselProvider>
    );
  }
}

export default Testimonials;
