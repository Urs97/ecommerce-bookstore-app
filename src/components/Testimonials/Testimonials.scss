@import "../../styles/shared.scss";

.testimonial-provider {
  width: 100%;
  // Height and padding in .testimonial have a wierd package synergy, flexbox doesnt work
  height: 17em;
  position: relative;

  .testimonial {
    width: 100%;
    height: 100%;
    padding: 3em $horizontal-content-padding;
    background-color: $dark-blue;
    color: #ffffff;

    .testimonial-slide .wrapper {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
    }

    .testimonial-slide img {
      width: 4em;
      height: 4em;
      border: 0.2em solid #ffffff;
      border-radius: 50%;
      object-fit: cover;
      margin-bottom: 1em;
    }

    .testimonial-slide p {
      font-size: 1.25em;
      margin-bottom: 1em;
      text-align: center;
      padding: 0 10em;
    }
  }

  .testimonials-btn {
    display: none;
    position: absolute;
    top: 50%;
    left: $horizontal-content-padding;
    -webkit-transform: translateY(-50%);
    -ms-transform: translateY(-50%);
    transform: translateY(-50%);
    background-color: rgba(0, 0, 0, 0);
    outline: none;
    border: none;
    text-decoration: none;
    padding: 1em 1em;

    span svg {
      color: #ffffff;
      font-size: 3em;
    }
  }

  .testimonials-btn:hover {
    background-color: rgba(0, 0, 0, 0.4);
  }

  .testimonials-btn:focus {
    outline: 0.2em solid #ffffff;
  }

  .testimonials-btn:not(.focus-visible) {
    outline: none;
  }

  .testimonials-btn.tstm-btn-next {
    right: $horizontal-content-padding;
    left: auto;
  }
}

.testimonial-provider:hover .testimonials-btn {
  display: inline-block;
}

@media (max-width: 1100px) {
  .testimonial-provider {
    .testimonial {
      padding: 3em $horizontal-content-padding-1100px;
    }
    .testimonials-btn {
      left: $horizontal-content-padding-1100px;
    }
    .testimonials-btn.tstm-btn-next {
      right: $horizontal-content-padding-1100px;
    }
  }
}

@media (max-width: 950px) {
  .testimonial-provider {
    .testimonial {
      padding: 3em $horizontal-content-padding-950px;
    }
    .testimonials-btn {
      left: $horizontal-content-padding-950px;
    }
    .testimonials-btn.tstm-btn-next {
      right: $horizontal-content-padding-950px;
    }
  }
}

// Mobile design

@media (max-width: 800px) {
  .testimonial-provider {
    height: 18em;
  }
}

@media (max-width: 710px) {
  .testimonial-provider {
    .testimonial {
      padding: 3em 1em;

      .testimonial-slide p {
        padding: 0 7em;
      }
    }

    .testimonials-btn {
      left: 2em;
    }
    .testimonials-btn.tstm-btn-next {
      right: 2em;
    }
  }
}

@media (max-width: 580px) {
  .testimonial-provider {
    .testimonial {
      .testimonial-slide p {
        padding: 0 5em;
      }
    }
    .testimonials-btn {
      left: 1.5em;
    }
    .testimonials-btn.tstm-btn-next {
      right: 1.5em;
    }
  }
}

@media (max-width: 500px) {
  .testimonial-provider {
    .testimonial {
      .testimonial-slide p {
        padding: 0 3em;
      }
    }
    .testimonials-btn {
      left: 0.5em;
    }
    .testimonials-btn.tstm-btn-next {
      right: 0.5em;
    }
  }
}

@media (max-width: 405px) {
  .testimonial-provider {
    height: 21em;
    .testimonial {
      .testimonial-slide p {
        padding: 0 3em;
      }
    }
    .testimonials-btn {
      left: 0.5em;
    }
    .testimonials-btn.tstm-btn-next {
      right: 0.5em;
    }
  }
}
