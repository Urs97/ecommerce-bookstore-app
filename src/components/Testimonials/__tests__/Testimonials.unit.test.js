import React from "react";
import { shallow } from "enzyme";
import Testimonials from "../Testimonials";

describe("Testimonials", () => {
  test("renders Testimonials", () => {
    const wrapper = shallow(<Testimonials />);
    expect(wrapper).toMatchSnapshot();
  });
});
