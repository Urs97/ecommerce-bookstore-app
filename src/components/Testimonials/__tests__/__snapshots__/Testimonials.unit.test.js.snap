// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials renders Testimonials 1`] = `
<n
  className="testimonial-provider"
  currentSlide={0}
  disableAnimation={false}
  disableKeyboard={false}
  dragEnabled={true}
  dragStep={1}
  hasMasterSpinner={true}
  infinite={true}
  interval={5000}
  isIntrinsicHeight={false}
  isPageScrollLocked={false}
  isPlaying={true}
  lockOnWindowScroll={false}
  naturalSlideHeight={9}
  naturalSlideWidth={16}
  orientation="horizontal"
  playDirection="forward"
  step={1}
  tag="div"
  totalSlides={3}
  touchEnabled={true}
  visibleSlides={1}
>
  <n
    className="testimonial"
  >
    <n
      className="testimonial-slide"
    >
      <div
        className="wrapper"
      >
        <n
          alt="Portrait photo by Štefan Štefančík on Unsplash"
          src="testimonial-portrait-1.jpg"
        />
        <p>
          I like Bookz more and more each day because it makes my life a lot easier. I'd really be lost without Bookz.
        </p>
        <span>
          Daniel Smith
        </span>
      </div>
    </n>
    <n
      className="testimonial-slide"
    >
      <div
        className="wrapper"
      >
        <n
          alt="Portrait photo by Robert Godwin on Unsplash"
          src="testimonial-portrait-2.jpg"
        />
        <p>
          I found what I needed and books arrived in a timely manner. The prices are super low, keep up the good work!
        </p>
        <span>
          Holly Dangerfield
        </span>
      </div>
    </n>
    <n
      className="testimonial-slide"
    >
      <div
        className="wrapper"
      >
        <n
          alt="Portrait photo by Mahesh Krishnamurthy on Unsplash"
          src="testimonial-portrait-3.jpg"
        />
        <p>
          It really saves me time and effort. Accesible book reading is exactly what I feel has been lacking so far and I have nothing but praise for the way Bookz goes about it!
        </p>
        <span>
          Steve Abrams
        </span>
      </div>
    </n>
  </n>
  <n
    className="testimonials-btn"
  >
    <span
      aria-label="left arrow icon"
      role="img"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              256,
              512,
              Array [],
              "f104",
              "M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z",
            ],
            "iconName": "angle-left",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </span>
  </n>
  <n
    className="testimonials-btn tstm-btn-next"
  >
    <span
      aria-label="right arrow icon"
      role="img"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              256,
              512,
              Array [],
              "f105",
              "M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z",
            ],
            "iconName": "angle-right",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </span>
  </n>
</n>
`;
